api/v1alpha3/condition_consts.go

```diff
@@ -25,3 +25,30 @@ const (
 )
 
 // ANCHOR_END: CommonConditions
+
+const (
+	// MachineHealthCheckSuccededCondition is set on machines that have passed a healthcheck by the MachineHealthCheck controller.
+	// In the event that the health check fails it will be set to False.
+	MachineHealthCheckSuccededCondition ConditionType = "HealthCheckSucceeded"
+
+	// HasFailureReason is the reason used when a machine has a FailureReason.
+	HasFailureReason = "MachineHasFailureReason"
```

Comment:
```suggestion
	HasFailureReason = "MachineHasFailure"
```

Reply:
I'm not sure if this is right. The field that's causing the MHC failure is `FailureReason`

Reply:
We should check both FailureReason and FailureMessage, because they're both optional. We can call it `MachineHasFailure` after that with an expanded comment explaining we're checking status for specific fields.

Reply:
Sounds good